Makefile

```diff
@@ -127,69 +100,194 @@ config: ## Set up default config
 
-.PHONY: start
-start:
-	sudo -E $(HOME)/.cargo/bin/auraed
+.PHONY: auraed-test
+auraed-test:
+	@$(cargo) test --target $(uname_m)-unknown-linux-musl -p auraed
 
-.PHONY: help
-help:  ## Show help messages for make targets
-	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(firstword $(MAKEFILE_LIST)) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[32m%-30s\033[0m %s\n", $$1, $$2}'
+.PHONY: auraed-test-all
+auraed-test-all:
+	@sudo -E $(cargo) test --target $(uname_m)-unknown-linux-musl -p auraed -- --include-ignored
+
+.PHONY: auraed-debug
+auraed-debug:
+	@$(cargo) install --target $(uname_m)-unknown-linux-musl --path ./auraed --debug --force
+
+.PHONY: auraed-release
+auraed-release:
+	@$(cargo) install --target $(uname_m)-unknown-linux-musl --path ./auraed --force
+
+#------------------------------------------------------------------------------#
+
+# AuraeScript Commands
+
+.PHONY: auraescript
+auraescript: proto auraescript-lint auraescript-debug ## Lint and install auraescript (for use during development)
+
+.PHONY: auraescript-lint
+auraescript-lint:
+	@$(cargo) clippy -p auraescript --all-features -- -D clippy::all -D warnings
+
+.PHONY: auraescript-test
+auraescript-test:
+	@$(cargo) test -p auraescript
+
+.PHONY: auraescript-test-all
+auraescript-test-all:
+	@$(cargo) test -p auraescript -- --include-ignored
+
+.PHONY: auraescript-debug
+auraescript-debug:
+	@$(cargo) install --path ./auraescript --debug --force
+
+.PHONY: auraescript-release
+auraescript-release:
+	@$(cargo) install --path ./auraescript --force
+
+#------------------------------------------------------------------------------#
+
+# aer Commands
+
+.PHONY: aer
+aer: proto aer-lint aer-debug ## Lint and install aer (for use during development)
+
+.PHONY: aer-lint
+aer-lint:
+	@$(cargo) clippy -p aer --all-features -- -D clippy::all -D warnings
+
+.PHONY: aer-test
+aer-test:
+	@$(cargo) test -p aer
+
+.PHONY: aer-test-all
+aer-test-all:
+	@$(cargo) test -p aer -- --include-ignored
+
+.PHONY: aer-debug
+aer-debug:
+	@$(cargo) install --path ./aer --debug --force
+
+.PHONY: aer-release
+aer-release:
+	@$(cargo) install --path ./aer --force
+
+#------------------------------------------------------------------------------#
+
+# Commands for other crates
+
+.PHONY: libs-lint
+libs-lint:
+	@$(cargo) clippy --all-features --workspace --exclude auraed --exclude auraescript --exclude aer  -- -D clippy::all -D warnings
+
+.PHONY: libs-test
+libs-test:
+	@$(cargo) test --workspace --exclude auraed --exclude auraescript --exclude aer
+
+.PHONY: libs-test-all
+libs-test-all:
+	@$(cargo) test --workspace --exclude auraed --exclude auraescript --exclude aer -- --include-ignored
+
+#------------------------------------------------------------------------------#
+
+# Commands for testing
+
+.PHONY: test-workflow
+test-workflow: ## Tests a github actions workflow locally using `act`
+	@act -W ./.github/workflows/$(file)
```

Comment:
where does `file` come from?  and do we need this to be a `make` command? can't i just run this myself if i need to test a workflow?

Reply:
🤷

Reply:
I added this, the idea was to run github actions tests using `make test-workflow file=001-cargo-install-ubuntu-make-install.yml` to test an individual file. If its too clunky it can be removed.